tests/UI/campaigns/functional/FO/09_productPage/03_writeReview.js

```diff
@@ -0,0 +1,334 @@
+Logout from FO
+Go to the BO and login
+Go to the "product comments" module configuration page
+Check if the review is visible in the "Reviews waiting for approval" table and if the content is correct
+Approve the review
+Check if the review is visible in the "Approved review" table and if the content is correct
+Logout from BO
+Go to the FO and login
+Go to product detail page
+Check if the review is visible and if the content is correct
+Go back to BO and delete approved review
+Go to FO and check if review is deleted
+ */
+describe('FO write a review', async () => {
+  // before and after functions
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+    page = await helper.newTab(browserContext);
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  describe('Go to FO product detail page and add a review', async () => {
+    it('should open the shop page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'openFoShop', baseContext);
+
+      await foHomePage.goTo(page, global.FO.URL);
+
+      const result = await foHomePage.isHomePage(page);
+      await expect(result).to.be.true;
+    });
+
+    it('should go to login page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToFOLoginPage', baseContext);
+
+      await foHomePage.goToLoginPage(page);
+
+      const pageHeaderTitle = await foLoginPage.getPageTitle(page);
+      await expect(pageHeaderTitle).to.equal(foLoginPage.pageTitle);
+    });
+
+    it('should login with default customer', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'signInFo', baseContext);
+
+      await foLoginPage.customerLogin(page, DefaultCustomer);
+
+      const isCustomerConnected = await foLoginPage.isCustomerConnected(page);
+      await expect(isCustomerConnected, 'Customer is not connected').to.be.true;
+    });
+
+    it('should go to home page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToHomePageToAddReview', baseContext);
+
+      await foHomePage.goToHomePage(page);
+
+      const result = await foHomePage.isHomePage(page);
+      await expect(result).to.be.true;
+    });
+
+    it('should go to the first product page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProductPage', baseContext);
+
+      await foHomePage.goToProductPage(page, 1);
+
+      const pageTitle = await productPage.getPageTitle(page);
+      await expect(pageTitle.toUpperCase()).to.contains(ProductData.firstProductData.name);
+    });
+
+    it('should get and save the actual comment count', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'getFoCommentCount', baseContext);
+
+      foCommentCount = await productPage.getNumberOfComment(page);
+      // TODO need assertion here ...
```

Comment:
```suggestion
      await expect(foCommentCount).to.be.at.least(0);
```

This will check if foCommentCount is still null or has a value.